ggshield/core/config/config.py

```diff
@@ -51,17 +77,23 @@ def config_path(self) -> Path:
 
     @property
     def instance_name(self) -> str:
+        return self.get_instance_name_and_source()[0]
+
+    def get_instance_name_and_source(self) -> Tuple[str, ConfigSource]:
         """
+        Return the instance name and source of the selected instance.
+
         The instance name (defaulting to URL) of the selected instance
         priority order is:
         - set from the command line (by setting cmdline_instance_name)
-        - env var (in auth_config.current_instance)
+        - GITGUARDIAN_INSTANCE env var
+        - GITGUARDIAN_API_URL env var
```

Comment:
We do not differentiate here between system env var or `.env` vars, since at this point, `load_dot_env` has been called so `.env` vars have overriden env vars.